### PR TITLE
Add bm25 reproduction script

### PR DESCRIPTION
**Pull request by dlrudwo1269**

### CODE REVIEW

File: examples/bm25_en_caption/run_bm25_baseline.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        (text_dir / p.name).symlink_to(p, target_is_directory=False)
+
+    for p in image_jsonl_paths:
+        (image_dir / p.name).symlink_to(p, target_is_directory=False)
+
+    print(f"Indexing text: {setting}{postfix}")
+    text_indexing_args = [
+        "-input", str(text_dir.resolve()),
+        "-collection", "JsonCollection",
+        "-index", f"indexes/lucene-index.atomic.text.flat.{setting}{postfix}",
```
Comment (dlrudwo1269) [inline examples/bm25_en_caption/run_bm25_baseline.py:98]: Provide absolute path for index

File: examples/bm25_en_caption/run_bm25_baseline.py [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        "-generator", "DefaultLuceneDocumentGenerator",
+        "-threads", "8", "-storePositions", "-storeDocvectors", "-storeRaw",
+    ]
+    anserini_index(text_indexing_args)
+
+    print(f"Indexing image: {setting}{postfix}")
+    image_indexing_args = [
+        "-input", str(image_dir.resolve()),
+        "-collection", "JsonCollection",
+        "-index", f"indexes/lucene-index.atomic.image.flat.{setting}{postfix}",
```
Comment (dlrudwo1269) [inline examples/bm25_en_caption/run_bm25_baseline.py:108]: Ditto

File: examples/bm25_en_caption/run_bm25_baseline.py [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+SPLITS = ["train", "validation", "test", "other"]
+SETTINGS = ["small", "base", "large"]
+ENCODING_FIELDS = ["text", "image_caption"]
+
+
+def get_args_parser():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--pyserini_path", required=True, type=str, help="Path to the Pyserini installation")
+    parser.add_argument("--anserini_fatjar_path", required=True, type=str, help="Path to the Anserini fatjar")
```
Comment (dlrudwo1269) [inline examples/bm25_en_caption/run_bm25_baseline.py:33]: Remove this

File: examples/bm25_en_caption/run_bm25_baseline.py [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from pathlib import Path
+import argparse
+
+from datasets import load_dataset
+from pyserini.setup import configure_classpath
```
Comment (dlrudwo1269) [inline examples/bm25_en_caption/run_bm25_baseline.py:20]: Importing `pyserini.pyclass.autoclass` will take care of this

**Comment by justram** [comment]
As per discussed in Slack, a few suggestions: 
1. Maybe use `pyserini.pyclass` to configure the anserini bindings automatically.
2. pyserini_path is not necessary
3. Add `pyserini` as requirement for the auto binding and simple python interface

File: requirements.txt
Comment (justram) [inline requirements.txt:1]: How about moving this file to `examples/bm25_en_caption`?

Comment (dlrudwo1269) [reply requirements.txt:1]: Yup sorry, meant to do that but think the file got misplaced. That makes sense.

**Pull request merged**